educator/src/Dscp/Educator/Web/Student/Types.hs

```diff
@@ -1,178 +1,119 @@
--- | Datatypes for Student HTTP API
+-- | Types specific to student API.
 
 module Dscp.Educator.Web.Student.Types
-       ( Student
-       , IsFinal (..)
-       , IsEnrolled (..)
-       , HasProof (..)
-       , Course (..)
-       , Assignment (..)
-       , Submission (..)
-       , Grade (..)
-       , BlkProof (..)
-       , NewSubmission (..)
-       , nsOwner
-
-       , _IsFinal
-       , assignmentTypeRaw
-       , liftAssignment
-       , liftSubmission
-       , signedSubmissionToRequest
-       , aDocumentType
-       ) where
-
-import Control.Lens (Iso', iso, makePrisms)
+    (
+      -- * Flags
+      IsEnrolled (..)
+
+      -- * Requests
+    , NewSubmission (..)
+    , nsOwner
+
+      -- * Responses
+    , CourseStudentInfo (..)
+    , AssignmentStudentInfo (..)
+    , SubmissionStudentInfo (..)
+    , saDocumentType
+
+      -- * Conversions
+    , studentLiftAssignment
+    , studentLiftSubmission
+    , signedSubmissionToRequest
+    ) where
+
 import Data.Aeson.Options (defaultOptions)
 import Data.Aeson.TH (deriveJSON)
-import Data.Time.Clock (UTCTime)
-import Servant (FromHttpApiData (..))
-
-import Dscp.Core (PrivateTx (..), addrFromText)
-import qualified Dscp.Core as Core
-import Dscp.Crypto (hash)
-import Dscp.Crypto (Hash, Raw)
-import Dscp.Util (Id, fromHex)
-import Dscp.Util.Aeson (AsHex)
-
-type Student = Core.Student
-
--- | Whether assignment is final in course.
-newtype IsFinal = IsFinal { unIsFinal :: Bool }
-    deriving (Eq, Show)
+import Servant (FromHttpApiData)
 
-makePrisms ''IsFinal
+import Dscp.Core
+import Dscp.Crypto
+import Dscp.Educator.Web.Types
+import Dscp.Util (Id)
 
 -- | Whether student is enrolled into a course.
 newtype IsEnrolled = IsEnrolled { unIsEnrolled :: Bool }
```

Comment:
Do we need this type? We protected from mixing it with any other `Bool`, but is it worth it?

Reply:
I feel bad when I see function of type `Bool -> Bool -> Bool -> m Something` (same endpoints would look like that), and making mistake here is easy and difficult to find, so I think it's worth it.

Reply:
Instead of creating newtype I can switch to `named` library, if it would look better

Reply:
In that case, usage of `newtype`s is justified.

Reply:
It's alright to use newtypes here imho. Especially if this type is a part of API.